WWDC/RelatedSessionsViewController.swift

```diff
@@ -54,36 +54,49 @@ struct RelatedSessionsView: View {
                 .lineLimit(1)
                 .truncationMode(.tail)
 
+            scrollView()
+        }
+        .frame(height: viewModel.sessions.isEmpty ? 0 : nil/*Metrics.height*/)
+        .opacity(viewModel.sessions.isEmpty ? 0 : 1)
+    }
+
+    func scrollView() -> some View {
+        GeometryReader { geometry in
             ScrollViewReader { scrollView in
                 ScrollView(.horizontal, showsIndicators: true) {
-                    SetScrollerStyle(.overlay)
-
-                    HStack(spacing: Metrics.itemSpacing) {
-                        ForEach(viewModel.sessions, id: \.identifier) { session in
-                            sessionButton(for: session, in: scrollView)
-                                .id(session.identifier)
+                    hStack(scrollView: scrollView)
+                        .background {
+                            SetScrollerStyle(.overlay)
                         }
-                    }
-                    .padding(.horizontal, 0)
-                    .padding(.bottom, Metrics.scrollerOffset)
+                        .padding(.bottom, Metrics.scrollerOffset)
                 }
+                .frame(maxWidth: geometry.size.width, alignment: .leading)
+                .frame(height: Metrics.itemHeight)
                 .onChange(of: viewModel.seed) { _, newValue in
-                    scrollView.scrollTo(viewModel.sessions.first?.identifier, anchor: .leading)
+                    scrollView.scrollTo(viewModel.sessions.first?.viewModel?.identifier, anchor: .leading)
                 }
             }
-            .frame(height: Metrics.scrollViewHeight)
-            .background(Color(.darkWindowBackground))
         }
-        .frame(height: viewModel.isHidden ? 0 : Metrics.height)
-        .opacity(viewModel.isHidden ? 0 : 1)
+        .frame(height: Metrics.scrollViewHeight)
+        .background(Color(.darkWindowBackground))
+        .padding(.top, Metrics.scrollerOffset)
+    }
+
+    func hStack(scrollView: ScrollViewProxy) -> some View {
+        LazyHStack(spacing: Metrics.itemSpacing) {
+            ForEach(viewModel.sessions, id: \.viewModel?.identifier) { session in
+                sessionButton(for: session, in: scrollView)
+            }
+        }
+        .frame(height: Metrics.itemHeight)
     }
 
-    func sessionButton(for session: SessionViewModel, in scrollView: ScrollViewProxy) -> some View {
+    func sessionButton(for session: SessionCellViewModel, in scrollView: ScrollViewProxy) -> some View {
         Button {
-            viewModel.selectSession(session)
+            viewModel.selectSession(session.viewModel!)
         } label: {
             SessionCellView(
-                cellViewModel: SessionCellViewModel(session: session),
```

Comment:
The outer VM needed to own this, SessionCellView is just @ObservedObject. Partly because of the table view.